### PR TITLE
chore(pkg): Fix permissions of vendored files

### PR DESCRIPTION
**Pull request by saulecabrera**
This commit ensures that all vendored files have the right permissions needed to build the gem.

The expected permissions are: `rw-r--r--`.

This issue was surfaced in
https://github.com/bytecodealliance/wasmtime-rb/actions/runs/4596038198/jobs/8117624399, in which a dependency update likely introduced an unexpected permission change in its files.

Locally inspecting the vendored files shows the discrepancy: the majority of the dependencies have the following permissions:

```
rw-r--r-- 1.2k saulecabrera  3 Apr 11:12 build.rs
.rw-r--r-- 1.5k saulecabrera  3 Apr 11:12 Cargo.toml
.rw-r--r--  869 saulecabrera  3 Apr 11:12 COPYRIGHT
.rw-r--r--  11k saulecabrera  3 Apr 11:12 LICENSE-APACHE
.rw-r--r--  12k saulecabrera  3 Apr 11:12 LICENSE-Apache-2.0_WITH_LLVM-exception
.rw-r--r-- 1.0k saulecabrera  3 Apr 11:12 LICENSE-MIT
.rw-r--r--  879 saulecabrera  3 Apr 11:12 README.md
drwxr-xr-x    - saulecabrera  3 Apr 11:12 src
```

While the failing dependencies have the following permissions:

```
.rw-r--r--  835 saulecabrera  3 Apr 11:12 Cargo.toml
.rw-r-----  11k saulecabrera  3 Apr 11:12 LICENSE-APACHE
.rw-r----- 1.1k saulecabrera  3 Apr 11:12 LICENSE-MIT
.rw-r----- 1.1k saulecabrera  3 Apr 11:12 README.md
drwxr-xr-x    - saulecabrera  3 Apr 11:12 src
drwxr-xr-x    - saulecabrera  3 Apr 11:12 tests
```

### CODE REVIEW

**Comment by saulecabrera** [comment]
Something's wrong with the Windows build -- I'll have to take a deeper look; I think it's very unlikely to be due to this PR though:


```
make: *** [Makefile:558: D:\a\wasmtime-rb\wasmtime-rb\target/release/wasmtime_rb.dll] Error 101
```

File: rakelib/pkg.rake [hunk 1]
```diff
@@ -75,9 +75,13 @@ namespace :pkg do
...
       sh "cargo vendor --versioned-dirs --locked #{vendor_dir} >> #{cargo_config_path} 2>/dev/null"
 
-      final_gemspec.files += dirglob("**/.cargo/**/*")
-      final_gemspec.files += dirglob("./#{vendor_dir}/**/*")
-      final_gemspec.files.reject! { |f| File.directory?(f) }
+      vendor_files = dirglob("./#{vendor_dir}/**/*").reject { |f| File.directory?(f) }      
+      # Ensure that all vendor files have the right read permissions,
+      # which are needed to build the gem. 
+      # The permissions that we want are `.rw-r--r--`
+      vendor_files.each { |f| File.chmod(0644, f) }
```
Comment (jbourassa) [inline rakelib/pkg.rake:82]: Have you considered something like `FileUtils.chmod("a+r", f)`? That way, we'd preserve existing permissions and only append to them, if e.g. a file was executable.

Looks like all RubyGem cares about is that [it's `& 0444`](https://github.com/ruby/ruby/blob/d2b87456f0e0046630dea9dd7fae4ec6324e9e92/lib/rubygems/specification_policy.rb#L232) (readable by user, group, other).

I don't know if that's desirable, but it's closer to the current behaviour.

Comment (saulecabrera) [reply rakelib/pkg.rake]: Added that in https://github.com/bytecodealliance/wasmtime-rb/pull/166/commits/4381070618fafe2b0c66e18a1af6c9e31e0cace8, and works as expected!

File: rakelib/pkg.rake [hunk 2]
```diff
@@ -75,9 +75,13 @@ namespace :pkg do
       mkdir_p ".cargo"
       sh "cargo vendor --versioned-dirs --locked #{vendor_dir} >> #{cargo_config_path} 2>/dev/null"
 
-      final_gemspec.files += dirglob("**/.cargo/**/*")
-      final_gemspec.files += dirglob("./#{vendor_dir}/**/*")
-      final_gemspec.files.reject! { |f| File.directory?(f) }
+      vendor_files = dirglob("./#{vendor_dir}/**/*").reject { |f| File.directory?(f) }
+      # Ensure that all vendor files have the right read permissions,
+      # which are needed to build the gem.
+      # The permissions that we want are `.rw-r--r--`
```
Comment (jbourassa) [inline rakelib/pkg.rake:81]: Nit: we want at least readable by all (`perm & 444 == 444`), it doesn't need to be exactly `rw-r--r--`

**Review by jbourassa — APPROVED** [review]
Thanks Saúl!

**Comment by ianks** [comment]
https://github.com/oxidize-rb/actions/releases/tag/v1.1.5

**Pull request merged**